src/core/mapping/mapping.cc

```diff
@@ -211,7 +211,8 @@ void StoreMapping::populate_layout_constraints(
     }
   } else
     fields.push_back(stores.front().get().region_field().field_id());
-  layout_constraints.add_constraint(Legion::FieldConstraint(fields, true /*contiguous*/));
+  layout_constraints.add_constraint(
```

Comment:
Why is this change needed? This seems like it could have major implications. Is it required in this PR?

Reply:
This and other changes for `tree_reduce` have been factored out into #699, so this diff will disappear once this PR is rebased on it. FYI, this diff actually fixes a performance problem for unbound stores; Legion's layout constraint check is currently too strict for single-field instances and it ends up refusing to reuse instances from unbound stores for downstream tasks even when they are reusable.